platform/mellanox/mlnx-platform-api/sonic_platform/pcie.py

```diff
@@ -0,0 +1,48 @@
+########################################################################
+#
+# Module contains a platform specific implementation of SONiC Platform
+# Base PCIe class
+#
+########################################################################
+import os
+import re
+
+try:
+    from sonic_platform.component import Component
+    from sonic_platform_base.sonic_pcie.pcie_common import PcieUtil
+except ImportError as e:
+    raise ImportError(str(e) + "- required module not found")
+
+
+class Pcie(PcieUtil):
+    # check the current PCIe device with config file and return the result
+    def get_pcie_check(self):
+        self.load_config_file()
+        for item_conf in self.confInfo:
+            id_conf = item_conf["id"]
+            dev_conf = item_conf["dev"]
+            fn_conf = item_conf["fn"]
+            bus_conf = self._bus_map.get(id_conf)
+            if bus_conf and self.check_pcie_sysfs(bus=int(bus_conf, base=16), device=int(dev_conf, base=16),
+                                                  func=int(fn_conf, base=16)):
+                item_conf["result"] = "Passed"
+            else:
+                item_conf["result"] = "Failed"
+        return self.confInfo
+
+    def _update_bus_map(self):
+        self._bus_map = {}
+        self.load_config_file()
+        dirs = os.listdir('/sys/bus/pci/devices/')
+        for dir in dirs:
+            m = re.search('....:(..):..\..', dir)
+            if m:
+                bus = m.group(1)
+
```

Comment:
else ???
what is your error flow here?
at least lets have a comment to describe it